### DISCUSSION TITLE
[Question] is there any way i can achieve exposure, contrast and saturation features in the library?

### DISCUSSION THREAD

**Discussion by sad-adnan** [hm21/pro_image_editor#226] [Q&A]
**is there any way i can achieve exposure, contrast and saturation features in the library?** 

@hm21 

if not can you please add this features.

**Comment by hm21** [comment]
Hi, I currently don't have enough time to add this feature, but I have already prepared everything for it. As you can see in the filter editor, you can also add custom Filters, where you can adjust settings such as Saturation, Brightness, or Contrast, as shown below:

```dart
configs: ProImageEditorConfigs(
  filterEditorConfigs: FilterEditorConfigs(filterList: [
    FilterModel(
      name: 'Saturation',
      filters: [
        ColorFilterAddons.saturation(0.5),
      ],
    ),
    FilterModel(
      name: 'Brightness',
      filters: [
        ColorFilterAddons.brightness(0.5),
      ],
    ),
    FilterModel(
      name: 'Contrast',
      filters: [
        ColorFilterAddons.contrast(0.5),
      ],
    ),
    FilterModel(
      name: 'Hue',
      filters: [
        ColorFilterAddons.hue(0.5),
      ],
    ),

    /// Default filters
    PresetFilters.none,
    PresetFilters.addictiveBlue,
    PresetFilters.addictiveRed,
    PresetFilters.aden,
    PresetFilters.amaro,
    PresetFilters.ashby,
    PresetFilters.brannan,
    PresetFilters.brooklyn,
    PresetFilters.charmes,
    PresetFilters.clarendon,
    PresetFilters.crema,
    PresetFilters.dogpatch,
    PresetFilters.earlybird,
    PresetFilters.f1977,
    PresetFilters.gingham,
    PresetFilters.ginza,
    PresetFilters.hefe,
    PresetFilters.helena,
    PresetFilters.hudson,
    PresetFilters.inkwell,
    PresetFilters.juno,
    PresetFilters.kelvin,
    PresetFilters.lark,
    PresetFilters.loFi,
    PresetFilters.ludwig,
    PresetFilters.maven,
    PresetFilters.mayfair,
    PresetFilters.moon,
    PresetFilters.nashville,
    PresetFilters.perpetua,
    PresetFilters.reyes,
    PresetFilters.rise,
    PresetFilters.sierra,
    PresetFilters.skyline,
    PresetFilters.slumber,
    PresetFilters.stinson,
    PresetFilters.sutro,
    PresetFilters.toaster,
    PresetFilters.valencia,
    PresetFilters.vesper,
    PresetFilters.walden,
    PresetFilters.willow,
    PresetFilters.xProII,
  ]),
)
```

This approach will work, but like in the filter editor, it only allows users to apply one configuration at a time. I recommend copying the filter editor and adjusting the logic slightly so that users can apply all configurations simultaneously.

**Reply by dev-basit-seerox** [reply]
i have added an icon in the bootom appbar 
on which three sliders are shown it is doing adjustment in the image , but when i move to filter or blur screen the orignal image is displayed not the adjusted 
but when i apply any filter or blur or crop to it and press the done both adjustment and filter or blur is applied on it...!
i want to update the image on fliter and other editors how can i achive it

 Widget _buildImage() {
    return Hero(
      tag: heroTag,
      createRectTween: (begin, end) => RectTween(begin: begin, end: end),
      child: _initialized
          ? TransformedContentGenerator(
        transformConfigs: stateManager.transformConfigs,
        configs: configs,
        child: Stack(
          children: [
            ColorFiltered(
              colorFilter: ColorFilter.matrix(_combinedMatrix),
              child: FilteredImage(
                width: sizesManager.decodedImageSize.width,
                height: sizesManager.decodedImageSize.height,
                configs: configs,
                image: editorImage,
                filters: stateManager.activeFilters,
                blurFactor: stateManager.activeBlur,
              ),
            ),
            // ColorFiltered(
            //   colorFilter: ColorFilter.matrix(_combinedMatrix),
            //   child: FilteredImage(
            //     width: sizesManager.decodedImageSize.width,
            //     height: sizesManager.decodedImageSize.height,
            //     configs: configs,
            //     image: editorImage,
            //     filters: stateManager.activeFilters,
            //     blurFactor: stateManager.activeBlur,
            //     colorFilterMatrix: _combinedMatrix,
            //   ),
            // ),
          ],
        ),
      )
          : AutoImage(
        editorImage,
        fit: BoxFit.contain,
        width: sizesManager.decodedImageSize.width,
        height: sizesManager.decodedImageSize.height,
        configs: configs,
      ),
    );
  }

**Comment by lcundiff** [comment]
I believe this has been added in 6.0.0 yes?

**Comment by hm21** [comment]
@dev-basit-seerox Apologies, I hadn't noticed your comment earlier. As @lcundiff mentioned, this feature is now included in version >= `6.0.0`.
